power/src/power.rs

```diff
@@ -83,6 +155,55 @@ impl TestKind {
                 let q = parse_i64(data, "q").unwrap();
                 Ok(TestKind::IncreaseMultipleRegression { rho, q })
             }
+            "ANCOVA" => {
+                let k = parse_i64(data, "k").unwrap();
+                let q = parse_i64(data, "q").unwrap();
+                let p = parse_i64(data, "p").unwrap();
+                Ok(TestKind::ANCOVA { k, q, p })
+            }
+            "oneWayANOVA" => {
+                let k = parse_i64(data, "k").unwrap();
+                Ok(TestKind::OneWayANOVA { k })
+            }
+            "twoWayANOVA" => {
+                let k = parse_i64(data, "k").unwrap();
+                let q = parse_i64(data, "q").unwrap();
+                Ok(TestKind::TwoWayANOVA { k, q })
+            }
+            "betweenRepeatedANOVA" => {
+                let k = parse_i64(data, "k").unwrap();
+                let m = parse_i64(data, "m").unwrap();
+                let rho = parse_f64(data, "rho").unwrap();
+                Ok(TestKind::BetweenRepeatedANOVA { k, m, rho })
+            }
+            "withinRepeatedANOVA" => {
+                let k = parse_i64(data, "k").unwrap();
+                let m = parse_i64(data, "m").unwrap();
+                let rho = parse_f64(data, "rho").unwrap();
+                let epsilon = parse_f64(data, "epsilon").unwrap();
+                if epsilon < (1.0 / (m as f64 - 1.0)) {
+                    Err(
+                        "lower bound of ε corresponds to 1 / (number of measurements - 1)"
+                            .to_string(),
+                    )
```

Comment:
Good for now. I've opened an issue about passing errors to the front end (https://github.com/poweranalyses-org/poweranalyses/issues/18).